### PR TITLE
Add entity browser support

### LINKED ISSUE #15: Add support for media field widgets with entity browser
Currently, we have a problem, when the configuration is cleaned-up the majority of entity browsers are removed too.

So we have to create entity browsers configuration with correct defined bundles.

### PR DESCRIPTION
**Pull request by mtodor**
Fixes #15

### CODE REVIEW

File: src/ConfigCreator.php [hunk 1]
```diff
@@ -176,13 +207,45 @@ public function create(): ConfigCreator {
           $this->eventDispatcher->dispatch(ConfigCreatorEvents::FIELD_CREATE, new ConfigCreatorFieldCreateEvent($field, $field_configuration));
         }
 
+        // Recalculate dependencies for display form after bundle is created.
```
Comment (chrfritsch) [inline src/ConfigCreator.php]: Can we move this to EntityTypeBase::postCreate. I think it's better suited there

File: src/ConfigCreator.php [hunk 2]
```diff
@@ -176,13 +207,45 @@ public function create(): ConfigCreator {
...
+        $form_display_entity = $this->entityTypeManager->getStorage('entity_form_display')->load(sprintf('%s.%s.%s', $entity_type, $bundle_id, 'default'));
+        $form_display_entity->calculateDependencies();
+        $form_display_entity->save();
+
         $testbuilder_entity_type->postCreate($bundle, $bundle_config);
       }
     }
 
+    $this->fixMissingConfiguration();
```
Comment (chrfritsch) [inline src/ConfigCreator.php]: Call this from the CreateConfigCommand

File: src/ConfigCreator.php [hunk 3]
```diff
@@ -176,13 +207,45 @@ public function create(): ConfigCreator {
...
+
     return $this;
   }
 
+  /**
+   * Discover and fix missing configuration.
+   */
+  protected function fixMissingConfiguration() {
+    $config_names = array_flip($this->configFactory->listAll());
+    foreach (array_keys($config_names) as $config_name) {
```
Comment (chrfritsch) [inline src/ConfigCreator.php]: Get rid of array_flip and array_keys

File: src/Plugin/TestsiteBuilder/ConfigImporter/EntityBrowser.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function importConfig(string $original, string $missing) {
+    $missing_config_name = ConfigName::createByFullName($missing);
+    $original_config_name = ConfigName::createByFullName($original);
+
+    if ($missing_config_name->getType() == 'entity_browser' && $original_config_name->getType() == 'entity_form_display') {
```
Comment (chrfritsch) [inline src/Plugin/TestsiteBuilder/ConfigImporter/EntityBrowser.php]: Could we get some comments for this, please. It's very difficult to understand

**Review by chrfritsch — DISMISSED** [review]
The general architecture is fine. It's only very hard to understand the entity browser plugin code. Some comments there would be very useful.

**Comment by mtodor** [comment]
All review comments are addressed.

**Review by dbosen — APPROVED** [review]
For future extensions of the plugins, we should investigate in using plugin deriver instead of all the foreach, foreach/foreach madness for creating plugin instances.
But this should do for now.

**Pull request merged**